test/Project.toml

```diff
@@ -5,6 +5,7 @@ CUDA_Runtime_jll = "76a88914-d11a-5bdc-97e0-2f5a05c973a2"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 Distributed = "8ba89e20-285c-5b6f-9357-94700520ee1b"
+Enzyme = "7da242da-08ed-463a-9acd-ee780be4f1d9"
```

Comment:
To make this into an extension, just add the following sections below the `[deps]` section:
```
[weakdeps]
EnzymeCore = "f151be2c-9106-41f4-ab19-57ee4f262869"

[extensions]
CUDAEnzymeCoreExt = "EnzymeCore"
```
And add to `[extras]`
```
EnzymeCore = "f151be2c-9106-41f4-ab19-57ee4f262869"
```
Then add the usual compat entry for `EnzymeCore` to `[compat]`. (see https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions))

This will make the code an extension on v1.9 and later. To also make the glue code work on older Julia versions, you could either:
1. make EnzymeCore a dependency of CUDA.jl for older Julia versions (for v1.9 and later, it will still only be a weak dependency; see https://pkgdocs.julialang.org/dev/creating-packages/#Transition-from-normal-dependency-to-extension)
2. since Requires is already a dependency here, you could conditionally load the extension file on older Julia versions (see https://pkgdocs.julialang.org/dev/creating-packages/#Requires.jl)

I'm happy to help set this up if given push rights to this branch.

Reply:
Added to the repo.